### DISCUSSION TITLE
Generate C# client via NSwag and/or NSwagStudio

### DISCUSSION THREAD

**Discussion by Guy-Heylens-Elia** [project-origin/wallet#197] [Ideas]
Hello, 

In order to facilitate the generation of a C# client it would be a good idea to expose the json or yaml file of the api spec https://app.swaggerhub.com/apis-docs/Project-Origin/WalletSystem/0.8.0#/ 

When using NSwagStudio you can point to the URL containing the json file, once you have that you can export/save your NSwagStudio settings as a .nswag file and store that in your repository.

With that .nswag file you can even further specify the conversions from one type to another.

![image](https://github.com/project-origin/wallet/assets/104833610/8aeb26de-1b67-4081-ad7b-1a845526cfbf)

**Comment by wisbech** [comment]
Would it make sense to make this a github action on deployment of each new version ? [actions/swagger-ui-action](https://github.com/marketplace/actions/swagger-ui-action)
